Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/network/NetworkStateProvider.kt

```diff
@@ -101,7 +104,14 @@ class NetworkStateProvider @Inject constructor(
         private val isFakeMeteredConnection: Boolean = false
     ) {
         val isMeteredConnection: Boolean
-            get() = isFakeMeteredConnection || !(capabilities?.hasCapability(NET_CAPABILITY_NOT_METERED) ?: false)
+            get() {
+                val unMetered = if (BuildVersionWrap.hasAPILevel(Build.VERSION_CODES.N)) {
+                    capabilities?.hasCapability(NET_CAPABILITY_NOT_METERED) ?: false
+                } else {
+                    capabilities?.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) ?: false
+                }
+                return isFakeMeteredConnection || !unMetered
+            }
```

Comment:
This is the key change.
The rest is testing and some refactoring because the `ApiLevel` pattern I added some time ago just is not useful enough. I replaced it with a mockable Kotlin object.